### PR TITLE
Feature/ipehr 43

### PR DESCRIPTION
**Pull request by DmitryLakhteen**

### CODE REVIEW

**Review by Setest — APPROVED** [review]

File: src/pkg/docs/service/ehr/ehr.go [hunk 1]
```diff
@@ -169,3 +153,24 @@ func (s EhrService) GetDocBySubject(userId, subjectId, namespace string) (docDec
...
+	if err != nil {
+		return
+	}
+
+	docDecrypted, err := s.Doc.GetDocFromStorageById(userId, docMeta.StorageId, []byte(ehrId))
+	if err != nil {
+		return
+	}
+
+	ehrDoc, err := s.ParseJson(docDecrypted)
```
Comment (1db1) [inline src/pkg/docs/service/ehr/ehr.go:168]: error handling needed

File: src/pkg/docs/service/ehr/ehr_status.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	key := chacha_poly.GenerateKey()
+
+	statusStorageId, err := s.saveStatusToStorage(status, key, ehrId)
+	if err != nil {
+		return err
+	}
+
+	subjectId := status.Subject.ExternalRef.Id.Value
+	subjectNamespace := status.Subject.ExternalRef.Namespace
+	err = s.Doc.SubjectIndex.AddEhrSubjectsIndex(ehrId, subjectId, subjectNamespace)
```
Comment (1db1) [inline src/pkg/docs/service/ehr/ehr_status.go:83]: error handling needed

File: src/pkg/docs/service/ehr/ehr.go [hunk 3]
```diff
@@ -169,3 +153,24 @@ func (s EhrService) GetDocBySubject(userId, subjectId, namespace string) (docDec
 	}
 	return
 }
+
+func (s *EhrService) UpdateDocumentStatus(userId, ehrId string, status model.EhrStatus) (err error) {
```
Comment (1db1) [inline src/pkg/docs/service/ehr/ehr.go:157]: it's better to pass a pointer to model.EhrStatus

**Comment by DmitryLakhteen** [comment]
FIxed

**Pull request merged**